### PR TITLE
signed messages in hexdump format, not base 58 - fixes #65

### LINKED ISSUE #65: [Hardware-Wallet] Don't use base58 format for signature
The hardware wallet sends the signature it computes in the base58 format. 
Just send them in hex format is fine.

### PR DESCRIPTION
**Pull request by stdevAlDen**
Fixes #65 

Changes (before this hw returns signed msg in base 58, this pr change it to hex format):
  - `deviceSignMessage` and `deviceSignMessage` messages returns the signed message in hex format.

  - `msgSignTransactionMessageImpl` encode in hex format.
  
- Refactor build scheme, now the firmware is build as a separate library and this can be linked against a main for tests or the main to be use in production with the firmware.
  
- Refcator in `fsm` it was split into two files, `fsm_impl` and `fsm` it selft, the funtions from `fsm_impl` return some kind of `err_code` and in general are more easy to use in unit tests.
  
 - Add some unit tests for `tiny-firmware` folder.

Does this change need to mentioned in CHANGELOG.md?
yes

Requires testing
yes

The `test_msgSkycoinSignMessageReturnIsInHex` test is not fully deterministic, it can return false positive in some specific cases for a broken feature as this check that each byte in the signed message have to be one of the base 16 valid digits (`[0-9], [a-f], [A-F]`).

### CODE REVIEW

File: tiny-firmware/firmware/fsm.c [hunk 1]
```diff
@@ -334,40 +334,32 @@ int fsm_getKeyPairAtIndex(uint32_t nbAddress, uint8_t* pubkey, uint8_t* seckey,
...
-	else
-	{
+	} else {
 		layoutRawMessage("Signature failed");
 	}
-	size_sign = sizeof(sign58);
-    b58enc(sign58, &size_sign, signature, sizeof(signature));
-	memcpy(resp->signed_message, sign58, size_sign);
+	const size_t hex_len = 2 * sizeof(signature);
+  char signature_in_hex[hex_len];
```
Comment (olemis) [inline tiny-firmware/firmware/fsm.c]: wrong indentation , are you combining tabs and whitespace?

Comment (stdevAlDen) [reply tiny-firmware/firmware/fsm.c]: > wrong indentation , are you combining tabs and whitespace?

I am using tabs with 4 characters length, this was a mistake... I'm going to solve this

**Comment by stdevAlDen** [comment]
> This PR needs equivalents in go and js repositories , plus tests .

Ok, I'm going to solve this [here](https://github.com/skycoin/hardware-wallet-js/issues/37)

File: tiny-firmware/firmware/main.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#include "trezor.h"
```
Comment (olemis) [inline tiny-firmware/firmware/main.c]: Add Skycoin license header.

File: tiny-firmware/firmware/fsm_impl.c [hunk 3]
```diff
@@ -0,0 +1,81 @@
+#include "fsm_impl.h"
```
Comment (olemis) [inline tiny-firmware/firmware/fsm_impl.c]: Add Skycoin license header.

File: skycoin-api/check_digest.c [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#include "check_digest.h"
```
Comment (olemis) [inline skycoin-api/check_digest.c]: Add Skycoin license header.

File: tiny-firmware/firmware/test_fsm.c [hunk 5]
```diff
@@ -0,0 +1,100 @@
+#include <stdio.h>
```
Comment (olemis) [inline tiny-firmware/firmware/test_fsm.c]: Add Skycoin license header

**Pull request merged**